bentoml/artifact/onnx_model_artifact.py

```diff
@@ -0,0 +1,178 @@
+    >>> iris = load_iris()
+    >>> X, y = iris.data, iris.target
+    >>> X_train, X_test, y_train, y_test = train_test_split(X, y)
+    >>> clr = RandomForestClassifier()
+    >>> clr.fit(X_train, y_train)
+
+    >>> # Convert into ONNX format
+    >>> from skl2onnx import convert_sklearn
+    >>> from skl2onnx.common.data_types import FloatTensorType
+    >>> initial_type = [('float_input', FloatTensorType([None, 4]))]
+    >>>
+    >>> onnx_model = convert_sklearn(clr, initial_types=initial_type)
+    >>> with open("rf_iris.onnx", "wb") as f:
+    >>>     f.write(onnx_model.SerializeToString())
+    >>>
+    >>>
+    >>> import numpy
+    >>> import bentoml
+    >>> from bentoml.artifact import OnnxModelArtifact
+    >>> from bentoml.handlers import DataframeHandler
+    >>>
+    >>> @bentoml.env(auto_pip_dependencies=True)
+    >>> @bentoml.artifacts([OnnxModelArtifact('model', backend='onnxruntime')])
+    >>> class OnnxIrisClassifierService(bentoml.BentoService):
+    >>>     @bentoml.api(DataframeHandler)
+    >>>     def predict(self, df):
+    >>>         input_data = df.to_numpy().astype(numpy.float32
+    >>>         input_name = self.artifacts.model.get_inputs()[0].name
+    >>>         output_name = self.artifacts.model.get_outputs()[0].name
+    >>>         return self.artifacts.model.run(
+    >>>                     [output_name], {input_name: input_data}
+    >>>                )[0]
+    >>>
+    >>> svc = OnnxIrisClassifierService()
+    >>>
+    >>> # Option one: pack with path to model on local system
+    >>> svc.pack('model', './rf_iris.onnx')
+    >>>
+    >>> # Option two: pack with ONNX model object
+    >>> # svc.pack('model', onnx_model)
+    >>>
+    >>> # Save BentoService
+    >>> svc.save()
+    """
+
+    def __init__(self, name, backend='onnxruntime'):
+        super(OnnxModelArtifact, self).__init__(name)
+        if backend not in SUPPORTED_ONNX_BACKEND:
+            raise BentoMLException(
+                f'{backend} runtime is currently not supported for OnnxModelArtifact'
+            )
+        self.backend = backend
+
+    def _model_file_path(self, base_path):
+        return os.path.join(base_path, self.name + '.onnx')
+
+    def pack(self, obj):  # pylint:disable=arguments-differ
+        return _OnnxModelArtifactWrapper(self, obj)
+
+    def load(self, path):
+        if self.backend == 'onnxruntime':
+            return _OnnxModelArtifactWrapper(self, path)
+        else:
+            raise NotImplementedError(
+                f'{self.backend} runtime is not supported at the moment'
+            )
+
+    @property
+    def pip_dependencies(self):
+        dependencies = []
+        if self.backend == 'onnxruntime':
+            dependencies.append('onnxruntime')
+        return dependencies
+
+
+class _OnnxModelArtifactWrapper(BentoServiceArtifactWrapper):
+    def __init__(self, spec, path):
+        super(_OnnxModelArtifactWrapper, self).__init__(spec)
+
+        self.path = path
```

Comment:
this is super confusing..  rename parameter to `path_or_model_proto`

here do

```
self._onnx_file_path = None
self._model_proto = None
if _is_onnx_model_file_path_like(path_or_model_proto)
     self._onnx_file_path = path_or_model_proto
else:
     try:
        import onnx
        if isinstance(self.path, onnx.ModelProto):
           self._model_proto = path_or_model_proto
        else:
           raise ..
     except ImportError:
        raise ...
```

In the save below, simply do:
```
if self._model_proto:

elif self._onnx_file_path:

else:
     raise ...
```
```

if self.